services-custom/dynamodb-enhanced/src/main/java/software/amazon/awssdk/enhanced/dynamodb/DynamoDbEnhancedAsyncClient.java

```diff
@@ -47,35 +54,353 @@
+     * }
+     * </pre>
+     *
+     * @param request A {@link BatchGetItemEnhancedRequest} containing keys grouped by tables.
+     * @return a publisher {@link SdkPublisher} with paginated results of type {@link BatchGetResultPage}.
+     */
     default SdkPublisher<BatchGetResultPage> batchGetItem(BatchGetItemEnhancedRequest request) {
         throw new UnsupportedOperationException();
     }
 
+    /**
+     * Retrieves items from one or more tables by their primary keys, see {@link Key}. BatchGetItem is a composite operation
+     * where the request contains one batch of {@link GetItemEnhancedRequest} per targeted table.
+     * <p>
+     * The additional configuration parameters that the enhanced client supports are defined
+     * in the {@link BatchGetItemEnhancedRequest}.
+     * <p>
+     * <b>Partial results</b>. A single call has restraints on how much
+     * data can be retrieved. If those limits are exceeded, the call will yield a partial result. This may also be the case if
+     * provisional throughput is exceeded or there is an internal DynamoDb processing failure. The enhanced client does not
+     * currently support retrieving information on partial results.
+     * <p>
+     * This operation calls the low-level DynamoDB API BatchGetItem operation. Consult the BatchGetItem documentation for
+     * further details and constraints as well as current limits of data retrieval.
+     * <p>
+     * <b>Note:</b> This is a convenience method that creates an instance of the request builder avoiding the need to create one
+     * manually via {@link BatchGetItemEnhancedRequest#builder()}.
+     * <p>
+     * Example:
+     * <pre>
+     * {@code
+     *
+     * SdkPublisher<BatchGetResultPage> batchResults = enhancedClient.batchGetItem(r -> r.addReadBatches(
+     *     ReadBatch.builder(FirstItem.class)
+     *              .mappedTableResource(firstItemTable)
+     *              .addGetItem(i -> i.key(key1))
+     *              .addGetItem(i -> i.key(key2))
+     *              .build(),
+     *     ReadBatch.builder(SecondItem.class)
+     *              .mappedTableResource(secondItemTable)
+     *              .addGetItem(i -> i.key(key3))
+     *              .build())).join();
+     * }
+     * </pre>
+     *
+     * @param requestConsumer a {@link Consumer} of {@link BatchGetItemEnhancedRequest.Builder} containing keys grouped by tables.
+     * @return a publisher {@link SdkPublisher} with paginated results of type {@link BatchGetResultPage}.
+     */
     default SdkPublisher<BatchGetResultPage> batchGetItem(Consumer<BatchGetItemEnhancedRequest.Builder> requestConsumer) {
         throw new UnsupportedOperationException();
     }
 
+    /**
+     * Puts or deletes multiple items in one or more tables. BatchWriteItem is a composite operation where the request contains
+     * one batch of {@link PutItemEnhancedRequest} or {@link DeleteItemEnhancedRequest} per targeted table.
```

Comment:
This wording makes it look a bit like each batch must exclusively contain PutItem requests or DeleteItem requests but not a mix (which as we know is not true). Maybe we should try and make that clearer.

Reply:
Agree, re-worded.